### PR TITLE
Moved babelfish related content from rdsutils extension to tdsutils

### PR DESCRIPTION
**Pull request by Suthapalli-Ramya-satya-vasavi-srija**
Signed-off-by: vasavi suthapalli <svasusri@amazon.com>

### Description

Previously, some babelfish code logic related to dropping and renaming has been put in rdsutils extension. Now, it has been moved to tdsutils part.
 
### Issues Resolved

BABEL-3457
### Testing scenarios covered
* **Use case based -** Tested and added in the description
* **Boundary conditions -** Tested whether the error is being thrown when not enable_drop_babelfish_role and successfully dropping when enable_drop_babelfish_role is enabled.
* **Arbitrary inputs -** Tested lower/upper/mixed case input for bbf created roles/users, db
* **Negative test cases -** Tested whether able to throw error why we try to alter/drop bbf roles/users, db
* **Minor version upgrade tests -** Not applicable
* **Major version upgrade tests -** Not applicable
* **Performance tests -** Not applicable
* **Tooling impact -** Not applicable
* **Client tests -** Not applicable
### Testing
Tested locally for rename/drop database need to connect to another database.
```
-- psql -U svasusri -d postgres
-- install a babelfish
You are now connected to database "postgres" as user "svasusri".
postgres=# create database demo;
CREATE DATABASE
postgres=# \c demo
You are now connected to database "demo" as user "svasusri".
demo=# CREATE EXTENSION IF NOT EXISTS "babelfishpg_tsql" CASCADE;
NOTICE:  installing required extension "uuid-ossp"
NOTICE:  installing required extension "babelfishpg_common"
CREATE EXTENSION
demo=# ALTER SYSTEM SET babelfishpg_tsql.database_name = 'demo';
ALTER SYSTEM
demo=# call sys.initialize_babelfish('svasusri');
CALL
demo=> SELECT pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

demo=> create user test_user with password '123' createdb;
CREATE ROLE
demo=> alter database demo owner to test_user;
ALTER DATABASE
demo=> \c  postgres test_user
You are now connected to database "postgres" as user "test_user".
postgres=> alter database demo rename to abcd;
ERROR:  cannot rename active babelfish database
postgres=> drop database demo;
ERROR:  cannot drop active babelfish database

-- check modification to an active babelfish is blocked
demo=> \c demo svasusri
You are now connected to database "demo" as user "svasusri".
demo=# ALTER SYSTEM SET babelfishpg_tsql.database_name = "none";
ALTER SYSTEM
demo=# SELECT pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

demo=# show babelfishpg_tsql.database_name;
 babelfishpg_tsql.database_name 
--------------------------------
 none
(1 row)

demo=# \c  postgres test_user
You are now connected to database "postgres" as user "test_user".
-- metadata not cleaned, expect an error
postgres=> drop database demo;
ERROR:  babelfish metadata not removed, please run remove_babelfish before dropping database

-- no longer an active babelifsh, allowed
postgres=> alter database demo rename to abcd; 
ALTER DATABASE

-- still not allowed
postgres=> drop database abcd;
ERROR:  babelfish metadata not removed, please run remove_babelfish before dropping database
postgres=> drop database ABCD;
ERROR:  babelfish metadata not removed, please run remove_babelfish before dropping database

-- clean metadata
postgres=> \c abcd svasusri
You are now connected to database "abcd" as user "svasusri".
abcd=# call sys.remove_babelfish();
NOTICE:  drop cascades to 4 other objects
DETAIL:  drop cascades to view master_dbo.sysdatabases
drop cascades to function master_dbo.xp_qv(nvarchar,nvarchar)
drop cascades to function master_dbo.xp_instance_regread(nvarchar,sysname,nvarchar,integer)
drop cascades to function master_dbo.xp_instance_regread(nvarchar,sysname,nvarchar,nvarchar)
NOTICE:  drop cascades to view tempdb_dbo.sysdatabases
NOTICE:  drop cascades to 4 other objects
DETAIL:  drop cascades to view msdb_dbo.sysdatabases
drop cascades to view msdb_dbo.syspolicy_system_health_state
drop cascades to function msdb_dbo.fn_syspolicy_is_automation_enabled()
drop cascades to view msdb_dbo.syspolicy_configuration
CALL

-- dropping allowed
abcd=# \c postgres test_user
You are now connected to database "postgres" as user "test_user".
postgres=> drop database abcd;
DROP DATABASE

-- test normal case again, to make sure it is not affected
postgres=> create database trail;
CREATE DATABASE
postgres=> alter database trail rename to trail2;
ALTER DATABASE
postgres=> drop database trail2;
DROP DATABASE
postgres=> 
```
### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tds/src/backend/tds/tdsutils.c [hunk 1]
```diff
@@ -541,3 +575,571 @@ TdsErrorContextCallback(void *arg)
...
+check_babelfish_renamerole_restrictions(char *role)
+{
+	if (sql_dialect == SQL_DIALECT_TSQL)
+		return;
+	if (is_babelfish_role(role))
+	{
+		pfree(role);	/* avoid mem leak */
+		ereport(ERROR,
+				(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
+				 errmsg("Babelfish-created users/roles cannot be dropped or altered outside of a Babelfish session")));
```
Comment (xhfanhe) [inline contrib/babelfishpg_tds/src/backend/tds/tdsutils.c:983]: Please add a test case for this (rename user/role)

File: contrib/babelfishpg_tds/src/backend/tds/tdsutils.c [hunk 2]
```diff
@@ -541,3 +575,571 @@ TdsErrorContextCallback(void *arg)
...
+check_babelfish_droprole_restrictions(char *role)
+{
+	if (sql_dialect == SQL_DIALECT_TSQL)
+		return;
+	if (is_babelfish_role(role))
+	{
+		pfree(role);	/* avoid mem leak */
+		ereport(ERROR,
+			(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
+				errmsg("Babelfish-created users/roles cannot be dropped or altered outside of a Babelfish session")));
```
Comment (xhfanhe) [inline contrib/babelfishpg_tds/src/backend/tds/tdsutils.c:884]: Please add a test case for this (drop user/role)

File: contrib/babelfishpg_tds/src/backend/tds/tdsutils.c [hunk 3]
```diff
@@ -541,3 +575,571 @@ TdsErrorContextCallback(void *arg)
...
+	babelfish_db_name = GetConfigOption("babelfishpg_tsql.database_name", true, false);
+
+	if (!babelfish_db_name) /* not defined */
+		return;
+
+	babelfish_db_id = get_database_oid(babelfish_db_name, true);
+	if (babelfish_db_id == target_db_id) /* rename active babelfish database */
+		ereport(ERROR,
+				(errcode(ERRCODE_OBJECT_IN_USE),
+				 errmsg("cannot rename active babelfish database")));
```
Comment (xhfanhe) [inline contrib/babelfishpg_tds/src/backend/tds/tdsutils.c:1029]: Please add a test case for this (rename db)

File: contrib/babelfishpg_tds/src/backend/tds/tdsutils.c [hunk 4]
```diff
@@ -541,3 +575,571 @@ TdsErrorContextCallback(void *arg)
...
+	bool        has_bbf_md = false;
+
+	babelfish_db_name = GetConfigOption("babelfishpg_tsql.database_name", true, false);
+	if (!babelfish_db_name) /* not define */
+		return;
+	babelfish_db_id = get_database_oid(babelfish_db_name, true);
+	if (babelfish_db_id == target_db_id) /* drop active babelfish database */
+		ereport(ERROR,
+				(errcode(ERRCODE_OBJECT_IN_USE),
+				 errmsg("cannot drop active babelfish database")));
```
Comment (xhfanhe) [inline contrib/babelfishpg_tds/src/backend/tds/tdsutils.c:1077]: Please add a test case for this (drop db)

File: contrib/babelfishpg_tds/src/backend/tds/tdsutils.c [hunk 5]
```diff
@@ -541,3 +575,571 @@ TdsErrorContextCallback(void *arg)
...
+
+		if (configs && is_babelfish_ownership_enabled(configs))
+			has_bbf_md = true;
+	}
+	systable_endscan(scan);
+	table_close(relsetting, AccessShareLock);
+	if (has_bbf_md)
+		ereport(ERROR,
+				(errcode(ERRCODE_OBJECT_IN_USE),
+				 errmsg("babelfish metadata not removed, please run remove_babelfish before dropping database")));
```
Comment (xhfanhe) [inline contrib/babelfishpg_tds/src/backend/tds/tdsutils.c:1114]: I'm not sure if this one is testable in automatic test without breaking the instance. If this cannot be tested in JDBC, please at least test this scenario locally and paste the record in PR/CR description.

**Review by xhfanhe — DISMISSED** [review]

**Review by macwoody — DISMISSED** [review]

**Review by HarshLunagariya — DISMISSED** [review]

**Review by shalinilohia50 — APPROVED** [review]

**Review by macwoody — APPROVED** [review]

**Pull request merged**